src/layout/rectangles.js

```diff
@@ -33,10 +32,15 @@ export default function(layoutStrategy) {
         }
 
         selection.each(function(data, index) {
+
+            var g = dataJoin(this, data);
+            g.call(component);
+
             // obtain the rectangular bounding boxes for each child
-            var childRects = data.map(function(d, i) {
-                var childPos = position(d, i);
-                var childSize = size(d, i);
+            var childRects = g[0].map(function(node, i) {
```

Comment:
I don't think I have. I suppose in d3 it's unusual to want to do a projection of nodes into data, it's normally the other way around.